library/alloc/src/boxed.rs

```diff
@@ -1330,6 +1331,10 @@ impl<T: Copy> From<&[T]> for Box<[T]> {
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "box_from_cow", since = "1.45.0")]
 impl<T: Copy> From<Cow<'_, [T]>> for Box<[T]> {
+    /// Converts a `Cow<'_, [T]>` into a `Box<[T]>`
+    ///
+    /// This conversion allocates on the heap
+    /// and performs a copy of `s`.
```

Comment:
> This conversion allocates on the heap and performs a copy of `s`.

- It only copies into a new allocation when the slice was borrowed. If it was owned, it uses the `From<Vec<T>> for Box<[T]>` implementation, which doesn't always reallocate.

- What does `s` refer to?

Reply:
> It only copies into a new allocation when the slice was borrowed. If it was owned, it uses the From<Vec<T>> for Box<[T]> implementation, which doesn't always reallocate.

I was spent too much time thinking about this. Ended up convincing myself that an extra allocation was required. Will fix up.

Reply:
> What does s refer to?

Copied the message from a pre-existing example too quickly.

Reply:
Fixed in e41d5ae65c0f51ca459064808961e6d59de6f6c4